### PR TITLE
Backwards compat changes to ensure that all tokens start with the study name

### PR DESCRIPTION
**Pull request by shankari**
Related server changes:
https://github.com/e-mission/e-mission-server/commit/b73d671fb48e8bead7ee40bdb3e45c3befd32330
https://github.com/e-mission/e-mission-server/commit/087076005d5a272937b6415f5fe2c091f1d5b11e
https://github.com/e-mission/e-mission-server/commit/6e9f4d2bf054f8534cee8743715ac18efa3c7c55

Related PR comments:
https://github.com/e-mission/e-mission-docs/issues/732#issuecomment-1179828277

Basically, we:
- infer the study name from the connect URL
- ensure that any new autogenerated tokens have the study prefix
- fix up old style tokens to start with the study prefix

Once all the old style tokens are converted over, we can remove the final change.
Once we change the study config to include the name, *and* support config updates, we can remove the first change.

### CODE REVIEW

**Comment by shankari** [comment]
Should we also ensure that any new tokens follow the correct format?
Let's do that next week, once we have a chance for the tokens to change over.
